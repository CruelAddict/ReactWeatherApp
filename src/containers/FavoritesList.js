import React, {Component} from 'react';
import {connect} from 'react-redux';
import Actions from "../modules/actions";
import FavoritesListHeader from '../components/FavoritesListHeader'
import FavoritesElements from './FavoritesElements'

const mapStateToProps = state => ({
    ...state
});

const mapDispatchToProps = dispatch => ({
    createFavorite: item => dispatch(Actions.createFavorite(item)),
});


class FavoritesList extends Component {
    constructor() {
        super();
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            favoritesInput: ""
        }
    }


    handleSubmit(event) {
        this.props.createFavorite(this.state.favoritesInput);
        event.preventDefault();
    }

    handleChange(event) {
        this.setState({favoritesInput: event.target.value})
    }

    render() {
        return (
            <div>
                <FavoritesListHeader
                    handleSubmit={this.handleSubmit}
                    handleChange={this.handleChange}
                    state={this.state}
                />
                <FavoritesElements/>
            </div>

        )
    }
}

export default connect(
    mapStateToProps, mapDispatchToProps)(
    FavoritesList
)
