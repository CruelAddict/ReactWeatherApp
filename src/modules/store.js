import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducer from './rootReducer'
import store from "../index";

const savedItems = localStorage.getItem('favorites') ? JSON.parse(localStorage.getItem('favorites')).map(value => {
        return {
            ...value,
            isLoading: true
        };
    }) : [];

const initialState = {
    favorites: {
        items: savedItems
    },
    geo: {
        permissionGranted: null,
        coordinates: null
    }
};

export default () => {
    return createStore(
        reducer,
        initialState,
        applyMiddleware(thunk)
    )
}
