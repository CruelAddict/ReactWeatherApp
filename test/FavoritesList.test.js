import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });

import FavoritesList from '../src/containers/FavoritesList';
import store from '../src/modules/store'
import Actions from "../src/modules/actions";



describe ('FavoritesList', () => {

    it('Renders if store is empty', () => {
        const panel = shallow(
            <FavoritesList
                store={store}
            />
        );
        expect(panel).toMatchSnapshot()
    });

    it('Renders if store has items', () => {
        store.dispatch(Actions.createFavorite("London")).then( () => {
            const panel = shallow(
                <FavoritesList
                    store={store}
                />
            );
            expect(panel).toMatchSnapshot()
        });
    });

});
