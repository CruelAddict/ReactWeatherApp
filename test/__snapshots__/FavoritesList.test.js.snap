// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList Renders if store has items 1`] = `
<ContextProvider
  value={null}
>
  <FavoritesList
    createFavorite={[Function]}
    favorites={
      Object {
        "items": Array [
          Object {
            "id": 0,
            "name": "London",
            "pushedToServer": false,
          },
        ],
      }
    }
    geo={
      Object {
        "coordinates": null,
        "permissionGranted": null,
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`FavoritesList Renders if store is empty 1`] = `
<ContextProvider
  value={null}
>
  <FavoritesList
    createFavorite={[Function]}
    favorites={
      Object {
        "items": Array [],
      }
    }
    geo={
      Object {
        "coordinates": null,
        "permissionGranted": null,
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
